sonic-xcvrd/xcvrd/xcvrd_utilities/port_mapping.py

```diff
@@ -1,19 +1,31 @@
+import threading
 from sonic_py_common import daemon_base
 from sonic_py_common import multi_asic
 from sonic_py_common.interface import backplane_prefix, inband_prefix, recirc_prefix
 from swsscommon import swsscommon
 
 SELECT_TIMEOUT_MSECS = 1000
+DEFAULT_PORT_TBL_MAP = [
+    {'CONFIG_DB': swsscommon.CFG_PORT_TABLE_NAME},
+    {'STATE_DB': 'TRANSCEIVER_INFO'},
+    {'STATE_DB': 'PORT_TABLE', 'FILTER': ['host_tx_ready']},
+]
 
 
 class PortChangeEvent:
+    # thread_local_data will be used to hold class-scope thread-safe variables
+    # (i.e. global variable to the local thread), such as
+    # thread_local_data.PORT_EVENT. thread_local_data.PORT_EVENT dict will be
+    # initialized in subscribe_port_update_event, and used to store the latest
+    # port change event for each key, to avoid duplicate event processing.
+    thread_local_data = threading.local()
```

Comment:
@longhuan-cisco can we have a much simpler solution? If you instantiate `PortChangeEvent` class per thread, it will be much easier as each instance will be accessed by their own thread context.

Reply:
@longhuan-cisco the problem is PortChangeEvent is being used like a global variable which is a bad choice. We should use object oriented approach which will scale better

Reply:
Yes, agreed. When I was writing the code for main PR (sff_mgr), I was thinking the same to go with object approach by change the global variable to object scope. But this would be a wider refactor (touching more code), including changing the caller side (cmis_mgr in xcvrd.py) and migrating those global functions (`subscribe_port_update_event` and `handle_port_update_event`) to class method/etc. Thus, at that time, I chose to go with the more self-contained and less-code-change solution, by simply changing the one single global variable to thread-safe without having to touching other part of the code.

Since now, we split the main PR and tracking this enhancement/fix as a separate PR, I think it's a good time to do this refactor. Will update accordingly.

Reply:
@prgeor I updated the diff to make the logic object oriented and get rid of all the global variables/functions.